pandas/_libs/tslibs/period.pyx

```diff
@@ -1152,46 +1152,89 @@ cdef int64_t period_ordinal_to_dt64(int64_t ordinal, int freq) except? -1:
 
 
 cdef str period_format(int64_t value, int freq, object fmt=None):
-    cdef:
-        int freq_group
-
     if value == NPY_NAT:
         return "NaT"
 
-    if isinstance(fmt, str):
-        # Encode using current locale, in case fmt contains non-utf8 chars
-        fmt = <bytes>util.string_encode_locale(fmt)
-
     if fmt is None:
-        freq_group = get_freq_group(freq)
-        if freq_group == FR_ANN:
-            fmt = b"%Y"
-        elif freq_group == FR_QTR:
-            fmt = b"%FQ%q"
-        elif freq_group == FR_MTH:
-            fmt = b"%Y-%m"
-        elif freq_group == FR_WK:
-            left = period_asfreq(value, freq, FR_DAY, 0)
-            right = period_asfreq(value, freq, FR_DAY, 1)
-            return f"{period_format(left, FR_DAY)}/{period_format(right, FR_DAY)}"
-        elif freq_group == FR_BUS or freq_group == FR_DAY:
-            fmt = b"%Y-%m-%d"
-        elif freq_group == FR_HR:
-            fmt = b"%Y-%m-%d %H:00"
-        elif freq_group == FR_MIN:
-            fmt = b"%Y-%m-%d %H:%M"
-        elif freq_group == FR_SEC:
-            fmt = b"%Y-%m-%d %H:%M:%S"
-        elif freq_group == FR_MS:
-            fmt = b"%Y-%m-%d %H:%M:%S.%l"
-        elif freq_group == FR_US:
-            fmt = b"%Y-%m-%d %H:%M:%S.%u"
-        elif freq_group == FR_NS:
-            fmt = b"%Y-%m-%d %H:%M:%S.%n"
-        else:
-            raise ValueError(f"Unknown freq: {freq}")
+        return _period_default_format(value, freq)
+    else:
+        if isinstance(fmt, str):
+            # Encode using current locale, in case fmt contains non-utf8 chars
+            fmt = <bytes>util.string_encode_locale(fmt)
+
+        return _period_strftime(value, freq, fmt)
 
-    return _period_strftime(value, freq, fmt)
+
+cdef str _period_default_format(int64_t value, int freq):
+    """A faster default formatting function leveraging string formatting."""
```

Comment:
faster than what?  i assume this refers to the implementation this replaces, but that wont be clear to future readers